--- debian/backuppc/etc/backuppc/config.pl	2018-02-18 15:47:01.023059724 +0100
+++ debian/backuppc/etc/backuppc/config.pl	2018-02-18 15:46:28.394935123 +0100
@@ -41,7 +41,10 @@
 #
 # Host name on which the BackupPC server is running.
 #
-$Conf{ServerHost} = 'XXXXXX';
+$ENV{'PATH'} = '/bin:/usr/bin';
+delete @ENV{'IFS', 'CDPATH', 'ENV', 'BASH_ENV'};
+$Conf{ServerHost} = `hostname`;
+chomp($Conf{ServerHost});
 
 #
 # TCP port number on which the BackupPC server listens for and accepts
@@ -275,7 +278,7 @@
 # Full path to various commands for archiving
 #
 $Conf{SplitPath} = '/usr/bin/split';
-$Conf{ParPath}   = '';
+$Conf{ParPath}   = '/usr/bin/par2' if -x '/usr/bin/par2';
 $Conf{CatPath}   = '/bin/cat';
 $Conf{GzipPath}  = '/bin/gzip';
 $Conf{Bzip2Path} = '/bin/bzip2';
@@ -325,7 +328,7 @@
 #
 # The BackupPC user.
 #
-$Conf{BackupPCUser} = '';
+$Conf{BackupPCUser} = 'backuppc';
 
 #
 # Important installation directories:
@@ -924,7 +927,7 @@
 #
 # This setting only matters if $Conf{XferMethod} = 'smb'.
 #
-$Conf{SmbClientPath} = '/usr/bin/smbclient';
+$Conf{SmbClientPath} = '';
 
 #
 # Command to run smbclient for a full dump.
@@ -1507,7 +1510,7 @@
 # Full path for ssh. Security caution: normal users should not
 # allowed to write to this file or directory.
 #
-$Conf{SshPath} = '/usr/bin/ssh';
+$Conf{SshPath} = '/usr/bin/ssh' if -x '/usr/bin/ssh';
 
 #
 # Full path for nmblookup. Security caution: normal users should not
@@ -1516,7 +1519,7 @@
 # nmblookup is from the Samba distribution. nmblookup is used to get the
 # netbios name, necessary for DHCP hosts.
 #
-$Conf{NmbLookupPath} = '/usr/bin/nmblookup';
+$Conf{NmbLookupPath} = '';
 
 #
 # NmbLookup command.  Given an IP address, does an nmblookup on that
@@ -1845,7 +1848,7 @@
 # handler this is either a plain name (eg: "admin") or a fully-qualified
 # name (eg: "admin@mydomain.com").
 #
-$Conf{EMailFromUserName} = '';
+$Conf{EMailFromUserName} = 'backuppc';
 
 #
 # Destination address to an administrative user who will receive a
@@ -1854,7 +1857,7 @@
 # handler this is either a plain name (eg: "admin") or a fully-qualified
 # name (eg: "admin@mydomain.com").
 #
-$Conf{EMailAdminUserName} = '';
+$Conf{EMailAdminUserName} = 'backuppc';
 
 #
 # Destination domain name for email sent to users.  By default
@@ -2015,7 +2018,7 @@
 # Full URL of the BackupPC_Admin CGI script, or the configured path
 # for SCGI.  Used for links in email messages.
 #
-$Conf{CgiURL} = 'http://XXXXXX/cgi-bin/BackupPC_Admin';
+$Conf{CgiURL} = 'http://'.$Conf{ServerHost}.'/backuppc/index.cgi';
 
 #
 # Full path to the rrdtool command.  If available, graphs of pool usage
@@ -2024,7 +2027,7 @@
 # Security caution: normal users should not allowed to write to this file
 # or directory.
 #
-$Conf{RrdToolPath} = '';
+$Conf{RrdToolPath} = '/usr/bin/rrdtool';
 
 #   
 # Language to use.  See lib/BackupPC/Lang for the list of supported
